hudi-client/src/main/java/org/apache/hudi/HoodieWriteClient.java

```diff
@@ -492,6 +501,11 @@ protected void postCommit(HoodieCommitMetadata metadata, String instantTime,
     }
   }
 
+  @Override
+  protected void updateOperationType(HoodieCommitMetadata metadata) {
```

Comment:
If we move the operationType instance to base-class, you no longer need this overridden method.